protocol/dubbo/pool.go

```diff
@@ -50,9 +50,20 @@ type gettyRPCClient struct {
 
 var (
 	errClientPoolClosed = perrors.New("client pool closed")
+	// for shared tcp connection for same ip and port
+	gettyRPCClients sync.Map
 )
 
 func newGettyRPCClientConn(pool *gettyRPCClientPool, protocol, addr string) (*gettyRPCClient, error) {
+
+	if client, ok := gettyRPCClients.Load(addr); ok {
+		cl := client.(*gettyRPCClient)
+		if cl.isAvailable() {
```

Comment:
I'm not sure that when client received the response, the connect will be closed.